crates/ty_python_semantic/src/types/cyclic.rs

```diff
@@ -1,23 +1,39 @@
 use crate::FxIndexSet;
 use crate::types::Type;
+use std::cmp::Eq;
+use std::hash::Hash;
 
-#[derive(Debug, Default)]
-pub(crate) struct TypeTransformer<'db> {
-    seen: FxIndexSet<Type<'db>>,
+pub(crate) type TypeTransformer<'db> = CycleDetector<Type<'db>, Type<'db>>;
+
+impl Default for TypeTransformer<'_> {
+    fn default() -> Self {
+        // TODO: proper recursive type handling
+
+        // This must be Any, not e.g. a todo type, because Any is the normalized form of the
+        // dynamic type (that is, todo types are normalized to Any).
+        CycleDetector::new(Type::any())
+    }
+}
```

Comment:
Implementing `Default` for a type alias like this means that you can do this -- but falling back to `Any` might not be what you want for an arbitrary `CycleDetector` instance; it only really makes sense for type transformers:

```rs
fn foo() {
    let x = CycleDetector::<Type, Type>::default();
}
```

Would it be better to implement `TypeTransformer` as a newtype rather than a type alias and implement `Deref`/`DerefMut`/`Default` for the newtype?

Reply:
OK!